instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/_server.py

```diff
@@ -136,74 +189,57 @@ def get_from_grpc_metadata(metadata, key) -> List[str]:
         else:
             yield
 
-    def _start_span(self, method):
-        span = self._tracer.start_as_current_span(
-            name=method, kind=trace.SpanKind.SERVER
+    def _start_span(self, handler_call_details, context):
+
+        attributes = {
+            "grpc.method": handler_call_details.method,
+        }
+
+        metadata = dict(context.invocation_metadata())
+        if "user-agent" in metadata:
+            attributes["grpc.user_agent"] = metadata["user-agent"]
+
+        # Split up the peer to keep with how other telemetry sources
+        # do it.  This looks like ipv6:[::1]:57284 or ipv4:127.0.0.1:57284.
+        host, port = context.peer().split(":", 1)[1].rsplit(":", 1)
+
+        # other telemetry sources convert this, so we will too
+        if host in ("[::1]", "127.0.0.1"):
+            host = "localhost"
+
+        attributes.update({"net.peer.name": host, "net.peer.port": port})
+
+        return self._tracer.start_as_current_span(
+            name=handler_call_details.method,
+            kind=trace.SpanKind.SERVER,
+            attributes=attributes,
+        )
+
+    def intercept_service(self, continuation, handler_call_details):
+        def telemetry_wrapper(behavior, request_streaming, response_streaming):
+            def telemetry_interceptor(request_or_iterator, context):
+
+                with self._set_remote_context(context):
+                    with self._start_span(
+                        handler_call_details, context
+                    ) as span:
+                        # wrap the context
+                        context = _OpenTelemetryServicerContext(context, span)
+
+                        # And now we run the actual RPC.
+                        try:
+                            return behavior(request_or_iterator, context)
+                        except Exception as error:
+                            # Bare exceptions are likely to be gRPC aborts, which
+                            # we handle in our context wrapper.
+                            # Here, we're interested in uncaught exceptions.
+                            # pylint:disable=unidiomatic-typecheck
+                            if type(error) != Exception:
+                                span.set_attribute("error", repr(error))
```

Comment:
Exceptions should be recorded as events in Otel. I think we should use `span.record_exception(error)` here

Reply:
Ah!  I didn't know that was a thing - will do.